File: src/main/kotlin/com/wtw/demo/comparator/BarDiffer.java
package com.wtw.demo.comparator;

import java.util.Optional;

public class BarDiffer implements Differ<Bar> {

    @Override
    public <S extends Bar> Optional<DiffResult> diff(S a, S b) {
        System.out.println("comparing bars..." + a + " <=> " + b);
        return Optional.of(DiffResult.change(a,b));
    }

}

File: src/main/kotlin/com/wtw/demo/comparator/BarDiffer.kt
package com.wtw.demo.comparator

import java.util.*

class BarDiffer : Differ<Bar> {
    override fun diff(a: Bar?, b: Bar?): Optional<DiffResult> {
        println("comparing bars..." + a.toString() + " <=> " + b)
        return Optional.of(DiffResult.change(a, b))    }

}
